src/Conf.ml

```diff
@@ -60,170 +78,356 @@ end
-          default
-      & info ["break-string-literals"] ~doc ~env)
+(** Options affecting formatting *)
+module Formatting = struct
+  let break_string_literals =
+    let doc =
+      "Break string literals. $(b,never) mode formats string literals as \
+       they are parsed, in particular, with escape sequences expanded. \
+       $(b,newlines) mode breaks lines at newlines. $(b,wrap) mode wraps \
+       string literals at the margin. Quoted strings such as \
+       $(i,{id|...|id}) are preserved. Can be set in a config file with a \
+       $(b,break-string-literals) = \
+       $(i,{)$(b,never)$(i,,)$(b,newlines)$(i,,)$(b,wrap)$(i,}) line."
+    in
+    let env = Arg.env_var "OCAMLFORMAT_BREAK_STRING_LITERALS" in
+    let name = "break-string-literals" in
+    let all =
+      [("wrap", `Wrap); ("newlines", `Newlines); ("never", `Never)]
+    in
+    let default = `Wrap in
+    C.choice ~name ~all ~default ~env ~doc ~allow_inline:true ~update:
+      (fun conf x -> {conf with break_string_literals= x} )
+
+  let doc_comments =
+    let doc =
+      "Doc comments position. Can be set in a config file with a \
+       `doc-comments = {before,after}` line."
+    in
+    let env = Arg.env_var "OCAMLFORMAT_DOC_COMMENTS" in
+    let name = "doc-comments" in
+    let all = [("after", `After); ("before", `Before)] in
+    let default = `After in
+    C.choice ~name ~all ~default ~env ~doc ~allow_inline:true ~update:
+      (fun conf x -> {conf with doc_comments= x} )
+
+  let escape_chars =
+    let doc =
+      "Escape encoding for character literals. Can be set in a config file \
+       with an `escape-chars = {decimal,hexadecimal,preserve}` line. \
+       `hexadecimal` mode escapes every character, `decimal` produces \
+       ASCII printable characters using decimal escape sequences as \
+       needed, and `preserve` escapes ASCII control codes but leaves the \
+       upper 128 characters unchanged."
+    in
+    let env = Arg.env_var "OCAMLFORMAT_ESCAPE_CHARS" in
+    let name = "escape-chars" in
+    let all =
+      [ ("decimal", `Decimal)
+      ; ("hexadecimal", `Hexadecimal)
+      ; ("preserve", `Preserve) ]
+    in
+    let default = `Preserve in
+    C.choice ~name ~all ~default ~env ~doc ~allow_inline:true ~update:
+      (fun conf x -> {conf with escape_chars= x} )
+
+  let escape_strings =
+    let doc =
+      "Escape encoding for string literals. Can be set in a config file \
+       with an `escape-strings = {decimal,hexadecimal,preserve}` line. See \
+       `--escape-chars` for the interpretation of the modes."
+    in
+    let env = Arg.env_var "OCAMLFORMAT_ESCAPE_STRINGS" in
+    let name = "escape-strings" in
+    let all =
+      [ ("decimal", `Decimal)
+      ; ("hexadecimal", `Hexadecimal)
+      ; ("preserve", `Preserve) ]
+    in
+    let default = `Preserve in
+    C.choice ~name ~all ~default ~env ~doc ~allow_inline:true ~update:
+      (fun conf x -> {conf with escape_strings= x} )
+
+  let break_infix =
```

Comment:
At one point the options were sorted alphabetically by name excluding any `no_` prefix. Maybe order them since this patch churns them all anyhow. And similarly for the order of the fields in the `t` record type above.

Reply:
sorted